src/pybkgmodel/data.py

```diff
@@ -501,23 +537,236 @@ def load_events(cls, file_name, cuts):
             event_data['pointing_az'],
             event_data['pointing_zd'],
             event_data['mjd'],
-            event_data['delta_t']
+            event_data['delta_t'],
+            None
         )
 
         return event_sample
 
+class DL3EventFile(EventFile):
+    """Reader for DL3 data file compliant with the GADF. For details see
+    https://gamma-astro-data-formats.readthedocs.io/
+
+    Parameters
+    ----------
+    file_name: str
+            Name of the DL3 file to use.
+    """
+    def __init__(self, file_name):
+        super().__init__(file_name)
+
+        self.file_name = file_name
+        self.obs_id = self.get_obs_id(file_name)
+        self.events = self.load_events(file_name)
+
+    @classmethod
+    def is_compatible(cls, file_name):
+        """Function checks whether the file is a fits file according to the file extension.
+
+        Parameters
+        ----------
+        file_name: str
+            Name of the DL3 file to use.
+
+        Returns
+        -------
+        bool
+            True if fits file according to file extension
+        """
+
+        ext = Path(file_name).suffixes
+        compatible = ".fits" in ext
```

Comment:
I don't think this is a robust way to check for compatibility, perhaps `astropy.fits` machinery is better?
https://docs.astropy.org/en/stable/io/fits/usage/verification.html

Reply:
In general I agree that it is not the most robust way. However, any deeper inspection of the files will require us to first define a strategy down to which level we like to check these files. While supporting the idea, maybe it going beyond the scope of this PR. What do you think? Maybe open a separate issue? Do you know whether gammapy already has a validation which we could use?

Reply:
For what concerns this specific line what I meant is that you are already importing here the `astropy.fits` module so doing something like

```python
try:
    with fits.open(input_file_path) as file:
        pass
    compatible = True
except OSError:
    compatible = False
```

should  be more reliable than just checking for the file extension (I didn't test it, but it might be that if the extension is not ".fits" a file can still a valid FITS file).

Technically `astropy` should already check some basic things (like e.g. the presence of a primary HDU).

For what concerns _gammapy_, if the FITS file in input is supposed to follow GADF I think it already checks for that schema, but I would not put too much in this PR.

Reply:
Adapted the solution and added it to the PR.

Reply:
Looks OK now.